api/spec/v1.yaml

```diff
@@ -3082,6 +3106,61 @@ components:
         verification_level:
           allOf: [$ref: '#/components/schemas/VerificationLevel']
 
+    EvmTokenSwap:
+      type: object
+      required: [pair_address]
+      properties:
+        pair_address:
+          type: string
+          description: |
+            The Oasis address of the swap pair contract.
+        pair_address_eth:
+          type: string
+          description: |
+            The Ethereum-compatible address of the swap pair contract.
+        factory_address:
+          type: string
+          description: |
+            The Oasis address of the swap factory contract.
+        factory_address_eth:
+          type: string
+          description: |
+            The Ethereum-compatible address of the swap factory contract.
+        token0_address:
+          type: string
+          description: |
+            The Oasis address of the first token in this swap.
+        token0_address_eth:
+          type: string
+          description: |
+            The Ethereum-compatible address of the first token in this swap
+        token1_address:
+          type: string
+          description: |
+            The Oasis address of the second token in this swap.
+        token1_address_eth:
+          type: string
+          description: |
+            The Ethereum-compatible address of the second token in this swap.
+        create_round:
+          type: integer
+          format: int64
+          description: |
+            The round when this swap pair was created.
+        reserve0:
+          allOf: [$ref: '#/components/schemas/TextBigInt']
+          description: |
+            The swap's liquidity pool of the first token, in that token's base units.
+        reserve1:
+          allOf: [$ref: '#/components/schemas/TextBigInt']
+          description: |
+            The swap's liquidity pool of the second token, in that token's base units.
```

Comment:
Is there any decimal information available? Or would the explorer just display in base units

Reply:
not from here, you'd have to get that from the token info

Reply:
Hm would it be possible to add it? (on a best-effort approach, since the `evm_tokens` analyzer might not have processed it or even run at all). The explorer will probably need this info, and would have to dispatch the second token info query afterwards which might be slow.

Reply:
ya I'll try joining this in

Reply:
added